From b9580f62225f5ad2e3d0ca5b3ff1788591479325 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Wed, 29 Mar 2023 15:25:29 -0700
Subject: [PATCH 1/7] Improve physical core count detection

Avoid erroring out when there is a problem reading /proc/cpuinfo and
instead just return 1.

Signed-off-by: William Douglas <william.douglas@intel.com>
---
 llvm/lib/Support/Unix/Threading.inc | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/llvm/lib/Support/Unix/Threading.inc b/llvm/lib/Support/Unix/Threading.inc
index 819748db4..bc219926c 100644
--- a/llvm/lib/Support/Unix/Threading.inc
+++ b/llvm/lib/Support/Unix/Threading.inc
@@ -340,9 +340,7 @@ static int computeHostNumPhysicalCores() {
   llvm::ErrorOr<std::unique_ptr<llvm::MemoryBuffer>> Text =
       llvm::MemoryBuffer::getFileAsStream("/proc/cpuinfo");
   if (std::error_code EC = Text.getError()) {
-    llvm::errs() << "Can't read "
-                 << "/proc/cpuinfo: " << EC.message() << "\n";
-    return -1;
+    return 1;
   }
   SmallVector<StringRef, 8> strs;
   (*Text)->getBuffer().split(strs, "\n", /*MaxSplit=*/-1,
--
2.42.1
